### DISCUSSION TITLE
S2 with  SHA256-HMAC hardware

### DISCUSSION THREAD

**Discussion by phil31** [BitMaker-hub/NerdMiner_v2#311] [General]
hi,

does it make sens to use the S2 version of the chip as it seem providing an hardware acceleratrion ? :
"This module provides hardware acceleration for SHA256-HMAC generation"
[https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/peripherals/hmac.html](url)

thanks, regards

**Comment by sany3001** [comment]
I would like to react, but provided link seems to be not working.

Update: I have googled. My knowledge is limited so I can only say I like the idea, but I do not know if HMAC can be used for NerdMiner needs.

**Reply by MyOwn2C** [reply]
Here is the S2 datasheet
https://www.espressif.com/sites/default/files/documentation/esp32-s2_datasheet_en.pdf

**Comment by phil31** [comment]
hi, simply the link to the Espressig foc !..
don't know why it's not working from Github .. copy/paste it ..

**Reply by sany3001** [reply]
Not important, but the link itself doesn't correspond to the link text. I would say not issue of GitHub.
Anyway: I am not expert in this topic. I can try to ask my colleague who is real ESP32 master. So if I will get something valuable, I will post it here.

**Comment by phil31** [comment]
i'm ESP expert, but dont nothing about crypto or mining / hashing stuffs !

**Reply by sany3001** [reply]
My colleague told me that it should be usable for our need. However we need to ask NerdMiner developers to do that.

**Comment by BitMaker-hub** [comment]
Hi Guys, just to explain you a little how NerdMiner evolved last year.

We started at 14Khs using only one thread + SHA cryptoAccelerator hardware
Then I noticed software implementations of sha were faster than SHA hardware,  so moved to C implementation of sha. (~29Khs)
Then we used other SHA lib, (>30Khs)
Then we used 2 threads (56Khs)
Then we used custom SHA implementation (78khs)

So, the cryptoAccelerator are incredibly slower than Software SHA + PSRAM

**Reply by sany3001** [reply]
Its a bit sad, but you made really great progress! Let's wait for some better ESP HW from Espressif. :-)

**Reply by phil31** [reply]
surprising results !
good job guys !

**Reply by BitMaker-hub** [reply]
@Sany3001 and @phil31. 
Can I reach you anywere? I've seen you've been very active and could be interesting to stay more connected via telegram on a devTeam

**Reply by sany3001** [reply]
You can find me on Telegram. However I hate those tools publishing private data like phone number. :-(
Is any chat group for NerdMiner existing on Telegram? If not, maybe it is time to create it.

Update: so "devTeam" shall be that group, right? There is some "DevTeam" I am already in.
May you can add this info into README.md.

**Reply by phil31** [reply]
very active !? only 1 question here ;o)

**Reply by BitMaker-hub** [reply]
ah hahah, @phil31 yes. hehe Sany is more active hehe

**Reply by sany3001** [reply]
[@phil31](https://github.com/phil31) with his first question which was really good one to me.

**Reply by sany3001** [reply]
According to me @PSLLSP provides lot of valuable inputs!
